docs/r1cs-docs-example.md

```diff
@@ -0,0 +1,530 @@
+The rank-1 constraint system API for programmatically defining constraint systems.
+
+## Building a proof-of-shuffle constraint system
+
+```ascii,no_run
+A shuffle is a permutation of a list of `k` scalars `{x_i}` into a list of `k` scalars `{y_i}`.
+
+Algebraically it can be expressed as a statement that for a free variable `z`, the roots of the two polynomials in terms of `z` are the same up to a permutation:
+
+    ∏(x_i - z) == ∏(y_i - z)
+
+The prover can commit to blinded scalars `x_i` and `y_i` then receive a random challenge `z`,
+and build a proof that the above relation holds.
+
+K-shuffle requires `2*(K-1)` multipliers.
+
+For K > 1:
+
+        (x_0 - z)---⊗------⊗---(y_0 - z)        // mulx[0], muly[0]
+                    |      |
+        (x_1 - z)---⊗      ⊗---(y_1 - z)        // mulx[1], muly[1]
+                    |      |
+                   ...    ...
+                    |      |
+    (x_{k-2} - z)---⊗      ⊗---(y_{k-2} - z)    // mulx[k-2], muly[k-2]
+                   /        \
+    (x_{k-1} - z)_/          \_(y_{k-1} - z)
```

Comment:
We could keep this diagram, but the rest of the block should probably be doc comments?

Reply:
👍